test/test_smartfile.py

```diff
@@ -44,7 +44,7 @@ def upload(self):
         f = StringIO(file_contents)
         f.seek(0)
         self.api.upload(TESTFN, f)
-        self.assertEquals(self.get_data()['size'], f.len)
+        self.assertEquals(self.get_data()['size'], f.tell())
```

Comment:
Does the `tell` function exist in python2 as well? Do we want to ensure backward compatibility correct?